src/utils.cpp

```diff
@@ -305,6 +305,81 @@ bool is_dir(const std::string& path)
   return S_ISDIR(buf.st_mode);
 }
 
+namespace {
+  class KernelHeaderTmpDir {
+    KernelHeaderTmpDir(const std::string& prefix) : path{prefix + "XXXXXX"}
+    {
+      if (::mkdtemp(&path[0]) == nullptr) {
+        throw std::runtime_error("creating temporary path for kheaders.tar.xz failed");
+      }
+    }
+
+    void move_to(const std::string& new_path)
+    {
+      int err = ::rename(path.c_str(), new_path.c_str());
+      if (err == 0) {
+        path = "";
+      }
+    }
+
+    std::string path;
+
+    friend std::string unpack_kheaders_tar_xz(const struct utsname&);
```

Comment:
Any reason not to instead make everything public?

Reply:
Yeah, I wondered that one myself when I went back to post v2 yesterday :)  That should shave off a few lines of weirdness.